src/examples/AddDataExample.java

```diff
@@ -0,0 +1,194 @@
+// This file is part of OpenTSDB.
+// Copyright (C) 2010-2012  The OpenTSDB Authors.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 2.1 of the License, or (at your
+// option) any later version.  This program is distributed in the hope that it
+// will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+// of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser
+// General Public License for more details.  You should have received a copy
+// of the GNU Lesser General Public License along with this program.  If not,
+// see <http://www.gnu.org/licenses/>.
+
+package net.opentsdb.examples;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+import com.stumbleupon.async.Callback;
+import com.stumbleupon.async.Deferred;
+
+import net.opentsdb.core.TSDB;
+import net.opentsdb.core.WritableDataPoints;
+import net.opentsdb.uid.NoSuchUniqueName;
+import net.opentsdb.uid.UniqueId.UniqueIdType;
+import net.opentsdb.utils.Config;
+
+/**
+ * Examples for how to add points to the tsdb.
+ * 
+ * @author Bryan Hernandez
+ * 
+ */
+public class AddDataExample {
+
+  public static void main(String[] args) throws Exception {
+
+    // Set these as arguments so you don't have to keep path information in
+    // source files
+    String pathToConfigFile = args[0]; // e.g.
+                                       // "/User/thisUser/opentsdb/src/opentsdb.conf"
+
+    // Create a config object with a path to the file for parsing. Or manually
+    // override settings.
+    // e.g. config.overrideConfig("tsd.storage.hbase.zk_quorum", "localhost");
+    Config config = new Config(pathToConfigFile);
+    final TSDB tsdb = new TSDB(config);
+
+    // Declare new metric
+    String metricName = "dummyFromjavaAPI";
+    // First check to see it doesn't already exist
+    byte[] byteMetricUID; // we don't actually need this for the first
+                          // .addPoint() call below.
+    // TODO: Ideally we could just call a not-yet-implemented tsdb.uIdExists()
```

Comment:
This whole bit is optional. If the user has autometric enabled, it will assign the UID in ``addPoint()``